### PR TITLE
(LTH-53) Fix handling of redirect on uploads with curl

### PR DESCRIPTION
**Pull request by pcarlisle**

### CODE REVIEW

**Comment by MikaelSmith** [comment]
The unit test appears to have problems with these changes.

**Comment by pcarlisle** [comment]
oh sweet i ran the wrong test suite

**Comment by pcarlisle** [comment]
ok

File: curl/src/client.cc [hunk 1]
```diff
@@ -261,6 +261,33 @@ namespace leatherman { namespace curl {
...
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKFUNCTION, seek_body);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+        result = curl_easy_setopt(_handle, CURLOPT_SEEKDATA, &ctx);
+        if (result != CURLE_OK) {
+            throw http_request_exception(ctx.req, curl_easy_strerror(result));
+        }
+
+        switch (method) {
```
Comment (MikaelSmith) [inline curl/src/client.cc]: This seems like it should be dealt with when you handle set_method.

Comment (pcarlisle) [inline curl/src/client.cc]: I can see the argument both ways, but keeping setting the size of the data with setting the data seemed more coherent to me.

Comment (pcarlisle) [inline curl/src/client.cc]: I'll change it if you feel strongly about it though.

Comment (MikaelSmith) [inline curl/src/client.cc]: I'm thinking it over. These two methods are closely related, but the alternatives are that they have to be called in a certain order, or we merge them into one. I'm ok with it the way it is.

**Comment by MikaelSmith** [comment]
Seems ok. :+1:

**Pull request merged**